include/sycl/helpers/sycl_buffers.hpp

```diff
@@ -108,7 +108,9 @@ make_buffer_impl(Iterator b, Iterator e, std::input_iterator_tag) {
   std::unique_ptr<type_> up{new type_[bufferSize]};
   std::copy(b, e, up.get());
   cl::sycl::buffer<type_, 1> buf(std::move(up), cl::sycl::range<1>(bufferSize));
+#ifdef __COMPUTECPP__
```

Comment:
OK, so ComputeCpp follows what I understand from the spec.
Great.
Just remove this ```buf.set_final_data(nullptr);```